### PR TITLE
Display hash_revision for `etcdctl endpoint hash`

### PR DESCRIPTION
**Pull request by ahrtr**
The server side change is done in https://github.com/etcd-io/etcd/pull/14537. 

This PR is the etcdctl side change, and also cover the test.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by ahrtr** [comment]
The server side change is done in https://github.com/etcd-io/etcd/pull/14537. 

This PR is the etcdctl side change, and also cover the test.

cc @spzala

**Comment by ahrtr** [comment]
Some examples output.

```
$ ./etcdctl endpoint hash --cluster -w table
+------------------------+------------+----------+
|        ENDPOINT        |    HASH    | REVISION |
+------------------------+------------+----------+
|  http://127.0.0.1:2379 | 2064120424 |       13 |
| http://127.0.0.1:22379 | 2064120424 |       13 |
| http://127.0.0.1:32379 | 2064120424 |       13 |
+------------------------+------------+----------+
```

```
$ ./etcdctl endpoint hash --cluster -w fields
"ClusterID" : 17237436991929493444
"MemberID" : 9372538179322589801
"Revision" : 13
"RaftTerm" : 2
"Endpoint" : "http://127.0.0.1:2379"
"Hash" : 2064120424
"HashRevision" : 13

"ClusterID" : 17237436991929493444
"MemberID" : 10501334649042878790
"Revision" : 13
"RaftTerm" : 2
"Endpoint" : "http://127.0.0.1:22379"
"Hash" : 2064120424
"HashRevision" : 13

"ClusterID" : 17237436991929493444
"MemberID" : 18249187646912138824
"Revision" : 13
"RaftTerm" : 2
"Endpoint" : "http://127.0.0.1:32379"
"Hash" : 2064120424
"HashRevision" : 13
```

```
$ ./etcdctl endpoint hash --cluster -w simple
http://127.0.0.1:2379, 2064120424, 13
http://127.0.0.1:22379, 2064120424, 13
http://127.0.0.1:32379, 2064120424, 13
```

```
$ ./etcdctl endpoint hash --cluster -w json | jq
[
  {
    "Endpoint": "http://127.0.0.1:2379",
    "HashKV": {
      "header": {
        "cluster_id": 17237436991929494000,
        "member_id": 9372538179322590000,
        "revision": 13,
        "raft_term": 2
      },
      "hash": 2064120424,
      "compact_revision": -1,
      "hash_revision": 13
    }
  },
  {
    "Endpoint": "http://127.0.0.1:22379",
    "HashKV": {
      "header": {
        "cluster_id": 17237436991929494000,
        "member_id": 10501334649042878000,
        "revision": 13,
        "raft_term": 2
      },
      "hash": 2064120424,
      "compact_revision": -1,
      "hash_revision": 13
    }
  },
  {
    "Endpoint": "http://127.0.0.1:32379",
    "HashKV": {
      "header": {
        "cluster_id": 17237436991929494000,
        "member_id": 18249187646912140000,
        "revision": 13,
        "raft_term": 2
      },
      "hash": 2064120424,
      "compact_revision": -1,
      "hash_revision": 13
    }
  }
]
```

**Comment by ahrtr** [comment]
Rebased this PR and updated the examples for `etcdctl endpoint hashkv` command in etcdctl readme.md.

**Comment by ahrtr** [comment]
cc @ptabor @serathius @spzala

File: etcdctl/README.md [hunk 1]
```diff
@@ -837,28 +837,73 @@ Prints a line of JSON encoding each endpoint URL and KV history hash.
...
 ```bash
-./etcdctl endpoint hashkv
-# 127.0.0.1:2379, 1084519789
+./etcdctl endpoint hashkv --cluster
+http://127.0.0.1:2379, 2064120424, 13
+http://127.0.0.1:22379, 2064120424, 13
+http://127.0.0.1:32379, 2064120424, 13
 ```
 
 Get the status for the default endpoint as JSON:
```
Comment (ptabor) [inline etcdctl/README.md:847]: Could you, please explain here what's the relationship between `hash_revision` and `revision`.
Why hash_revision can be stale and inconsistent with other members.

File: etcdctl/ctlv3/command/printer.go [hunk 2]
```diff
@@ -233,11 +233,12 @@ func makeEndpointStatusTable(statusList []epStatus) (hdr []string, rows [][]stri
 }
 
 func makeEndpointHashKVTable(hashList []epHashKV) (hdr []string, rows [][]string) {
-	hdr = []string{"endpoint", "hash"}
+	hdr = []string{"endpoint", "hash", "revision"}
```
Comment (ptabor) [inline etcdctl/ctlv3/command/printer.go:236]: maybe let's be consistent about all presentation forms and call it `hash_revision` ?

File: etcdctl/README.md [hunk 3]
```diff
@@ -837,28 +837,73 @@ Prints a line of JSON encoding each endpoint URL and KV history hash.
...
-+------------------------+------------+
-| http://127.0.0.1:2379  | 1084519789 |
-| http://127.0.0.1:22379 | 1084519789 |
-| http://127.0.0.1:32379 | 1084519789 |
-+------------------------+------------+
+$ ./etcdctl endpoint hash --cluster -w table
++------------------------+------------+----------+
+|        ENDPOINT        |    HASH    | REVISION |
++------------------------+------------+----------+
+|  http://127.0.0.1:2379 | 2064120424 |       13 |
```
Comment (ptabor) [inline etcdctl/README.md:903]: nit: The space ahead of 'http' seems nit needed (breaks aligment). Aren't we aligning to left.

**Review by ptabor — APPROVED** [review]
Thank you. Just a few nits.

File: etcdctl/README.md (hunk 1 above)
Comment (ahrtr) [reply etcdctl/README.md:847]: `hash_revision` is the revision up to which the hash is calculated. `revision` included in the response header is the latest revision the member has. Usually `revision` may be bigger than `hash_revision` when the cluster is under high load. 

`./etcdctl endpoint hashkv --cluster` just get each member's hash one by one, so the hash_revision might be inconsistent. Previously I was thinking to pre-fetch the revision, and then get hash using the same revision. But etcdctl is a CLI tool, and I expect users retry the command with a revision when they see inconsistent hash_revision

File: etcdctl/ctlv3/command/printer.go (hunk 2 above)
Comment (ahrtr) [reply etcdctl/ctlv3/command/printer.go]: Makes sense & updated.

File: etcdctl/README.md (hunk 3 above)
Comment (ahrtr) [reply etcdctl/README.md]: It's [ALIGN_RIGHT](https://github.com/etcd-io/etcd/blob/590911dfe36a972912a8f8ebb593614f00ce1f7e/etcdutl/etcdutl/printer_table.go#L34) by default, and I did not change it in this PR.

**Pull request merged**